recursive_metadata/write.py

```diff
@@ -0,0 +1,157 @@
+import os
+import sys
+import logging
+import json
+from dataclasses import dataclass
+from avro.schema import RecordSchema
+from importlib.metadata import version
+from datahub.metadata.schema_classes import (
+    KEY_ASPECTS, 
+    ASPECT_NAME_MAP,
+)
+from datahub.utilities.urns.urn import Urn, guess_entity_type
+from datahub.ingestion.graph.client import (
+    DataHubGraph,
+    get_default_graph
+)
+
+
+@dataclass
+class DataHubRegistryEntityEntry:
+   # Category of the DataHub Entity 
+   category: str
+   # Name of the key aspect for the DataHub Entity
+   key: str
+   # List of aspects that make up the DataHub Entity
+   aspects: list[str]
+
+
+@dataclass
+class DataHubIndex:
```

Comment:
Do you wish to just merge the `datahub_metadata_model` in this example and just add in the `README` for this? Otherwise 2 places to maintain. 

Not a hard blocker. Suggestion. I guess these are scripts so not a blocker

Reply:
I want to split them out. This file is an example of why and how to use the metadata model registry.